h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoModel.java

```diff
@@ -177,16 +186,90 @@ public static double scoreTree(byte[] tree, double[] row, int nclasses, boolean
         }
     }
 
-    public static String getDecisionPath(double leafAssignment) {
+    public interface DecisionPathTracker<T> {
```

Comment:
This is well designed. I like this approach of single tracker per walk.Easy to understand & use.